pkg/proxy/iptables/proxier.go

```diff
@@ -955,6 +955,11 @@ func (proxier *Proxier) syncProxyRules() {
 		proxier.endpointChainsNumber += len(proxier.endpointsMap[svcName])
 	}
 
+	nodeAddresses, nodeAddressesErr := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
```

Comment:
+1

Reply:
previously, in L1198-1201, if there was an error it `continue` . 
I think that `GetNodeAddress()` always return an empty `nodeAddress` if there is an error, so that should not change.
But it may happen that there is no error but we don't have any `nodeAddress`, is ok to `continue` in that case and skip the code that does not depend on `nodeAddress`?

Reply:
I would consider it a bug if `GetNodeAddresses` returns a nil error and no nodeAddresses. If it did, we wouldn't bind the node port locally. Maybe we should update GetNodeAddress to return an error in the end if len(uniqueAddressList) == 0?

Reply:
FYI https://github.com/kubernetes/kubernetes/pull/91461